### ISSUE TITLE
Inconsistency on the Dataset document

### ISSUE THREAD

**Issue by zasdfgbnm** [tensorflow/tensorflow#17135]
### System information
N/A

### Describe the problem
There are some inconsistencies in the document about Dataset.

#### Problem 1
In https://www.tensorflow.org/get_started/premade_estimators, i.e. the link to the most recent stable release (v1.5 for now), in "Getting the sample code" section, there is something that reads:

> The program described in this document is premade_estimator.py. This program uses iris_data.py To fetch its training data.

The links of `premade_estimator.py` and `iris_data.py` in the above sentence are to the master branch, instead of the branch corresponding to the correct version. It is not a good idea to point to the master branch because:

If you follow the link of  `iris_data.py`, you can see https://github.com/tensorflow/models/blob/master/samples/core/get_started/iris_data.py#L39
`return dataset` which is not the same as the `return dataset.make_one_shot_iterator().get_next()` in the document. This confuses the users.

To fix this, I would suggest to change the script that generates the web page to automatically generate links pointing to the correct github tag.

#### Problem 2
In https://www.tensorflow.org/versions/master/get_started/premade_estimators
i.e. the "Getting Started with TensorFlow" for the master branch,  the example code has already been changed to return a `Dataset`, which is consistent with https://github.com/tensorflow/models/blob/master/samples/core/get_started/iris_data.py#L39
This is good here.

But in https://www.tensorflow.org/versions/master/get_started/datasets_quickstart
the sample code still returns `dataset.make_one_shot_iterator().get_next()`

I would suggest making it consistent by returning a `Dataset` in the above code.  And if needed, consider adding something like:
> Since version 1.5, `Estimator` now supports `Dataset: input_fn` to return a `Dataset` instead of `Tensors`. So `return dataset.make_one_shot_iterator().get_next()` can be simplified to `return dataset`

### Source code / logs
N/A

**Comment by zasdfgbnm** [comment]
............
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: N/A
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: N/A
- **TensorFlow installed from (source or binary)**: N/A
- **TensorFlow version (use command below)**: N/A
- **Python version**: N/A
- **Bazel version (if compiling from source)**: N/A
- **GCC/Compiler version (if compiling from source)**: N/A
- **CUDA/cuDNN version**: N/A
- **GPU model and memory**: N/A
- **Exact command to reproduce**: N/A

**Comment by bignamehyp** [comment]
iris_data.py needs an update. @nealwu please take a look. Thanks

**Comment by nealwu** [comment]
Re-assigning to @MarkDaoust for the samples/ folder.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 15 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 77 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 93 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 109 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 124 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @MarkDaoust: It has been 139 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by MarkDaoust** [comment]
A chunk of this is fixed.

The "pointing to the correctly versioned branch" will be fixed as part of the migration of tensorflow docs to their own repo (coming soon).
